src/query/scan_manager.c

```diff
@@ -8041,20 +8110,41 @@ check_hash_list_scan (LLIST_SCAN_ID * llsidp, int *val_cnt, int hash_list_scan_y
       if (((vtype1 == DB_TYPE_OBJECT || vtype1 == DB_TYPE_VOBJ) && vtype2 == DB_TYPE_OID) ||
 	  ((vtype2 == DB_TYPE_OBJECT || vtype2 == DB_TYPE_VOBJ) && vtype1 == DB_TYPE_OID))
 	{
-	  return false;
+	  return HASH_METH_NOT_USE;
+	}
+      if (vtype1 != vtype2)
+	{
+	  llsidp->hlsid.need_coerce_type = true;
 	}
       build = build->next;
       probe = probe->next;
     }
   /* The number of probe regu_var and build regu match */
   if (build != NULL || probe != NULL)
     {
-      return false;
+      return HASH_METH_NOT_USE;
     }
   *val_cnt = build_cnt;
 
   /* 6. list file from dptr is not allowed */
   /* Since dptr is searched after scan_open_scan, it is checked when llsidp->list_id->tuple_cnt <= 0 */
 
-  return true;
+  /* list file size check */
+  if ((UINT64) llsidp->list_id->page_cnt * DB_PAGESIZE <= mem_limit)
+    {
+      return HASH_METH_IN_MEM;
+    }
+  else if ((UINT64) llsidp->list_id->tuple_cnt * (sizeof(HENTRY_HLS) + sizeof(QFILE_TUPLE_SIMPLE_POS)) <= mem_limit)
+    {
+      /* bytes of 1 row = sizeof(HENTRY_HLS) + sizeof(QFILE_TUPLE_SIMPLE_POS) = 36 bytes (64bit) */
+      /* HENTRY_HLS = pointer(8bytes) * 3 = 24 bytes*/
+      /* SIMPLE_POS = pageid(4bytes) + voldid(2bytes) + padding(2bytes) + offset(4bytes) = 12 bytes*/
+      return HASH_METH_HYBRID;
+    }
+  else
+    {
+      return HASH_METH_NOT_USE;
+    }
+
+  return HASH_METH_NOT_USE;
```

Comment:
this code is not reached, no need "else", simply "return HASH_METH_NOT_USE"

Reply:
I wanted to make the if-else more clear and put the return 'NOT_USE' at the end as defense code. If it doesn't matter, I'll keep it.